pkg/util/vsphere/vm/guest_id.go

```diff
@@ -0,0 +1,54 @@
+// Copyright (c) 2024 VMware, Inc. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+package vm
+
+import (
+	"fmt"
+	"strings"
+
+	vimtypes "github.com/vmware/govmomi/vim25/types"
+
+	vmopv1 "github.com/vmware-tanzu/vm-operator/api/v1alpha3"
+	"github.com/vmware-tanzu/vm-operator/pkg/conditions"
+	pkgctx "github.com/vmware-tanzu/vm-operator/pkg/context"
+)
+
+// GuestIDProperty is the property name for the guest ID in the config spec.
+const GuestIDProperty = "configSpec.guestId"
+
+// UpdateVMGuestIDCondition updates the VM's GuestID condition to false if the
+// taskInfo contains an InvalidArgument fault with an invalid guest ID property.
+// Otherwise, it updates the condition to true.
+func UpdateVMGuestIDCondition(
+	vmctx pkgctx.VirtualMachineContext,
+	configSpec vimtypes.VirtualMachineConfigSpec,
+	taskInfo *vimtypes.TaskInfo) {
+	// Return early if the current configSpec does not have a guest ID.
+	if configSpec.GuestId == "" {
```

Comment:
Shouldn't the condition be deleted here?

Reply:
I was trying to avoid resetting the condition when the VM is reconfigured without the guestID set in the config spec. Thinking it through now, that should only happen when vm.spec.guestID is removed. So you're right, we should remove the condition here. I will update the implementation and add a test case for this. Thanks for catching this!

Reply:
Also updated it to remove the condition when there is no error, instead of marking it as true, which will eventually be removed in a future VM reconfiguration anyways.